### PR TITLE
Add initial VDR generation pipeline

### PR DESCRIPTION
**Pull request by Scanteianu**
Migration from https://github.com/Scanteianu/openjdk-cve-parser/pull/2, which in turn is a fork of https://github.com/Scanteianu/openjdk-cve-parser.

Stuff in the data directory can be ignored, it's essentially a cache (except for vdr.json, which is the final output)

### CODE REVIEW

**Review by smlambert — COMMENTED** [review]
Thanks @Scanteianu !  

Perhaps remove the cached .json files from the data folder so they do not get checked in with the rest of these changes, if they are just a side-effect of a local run.  Can also add a .gitignore file so that when someone is working on PRs for this repository and running tests locally, they do not accidentally include working / cached files in their pull requests.

**Comment by Scanteianu** [comment]
> Thanks @Scanteianu !
> 
> Perhaps remove the cached .json files from the data folder so they do not get checked in with the rest of these changes, if they are just a side-effect of a local run. Can also add a .gitignore file so that when someone is working on PRs for this repository and running tests locally, they do not accidentally include working / cached files in their pull requests.

Thanks @smlambert , should i keep the downloads of the website? (the htmls)? In an ideal world i'd put all this stuff in some kind of database, but i'm not sure we need that much complexity yet. The goal is to avoid putting load on nist/ojvg every time we run the tool

File: README.md [hunk 1]
```diff
@@ -1,2 +1,20 @@
...
+Currently the VDR is generated by aggregating data from 2 sources: the OpenJDK Vulnerability Group and the NIST API.
+We use `ojvg_download.py` to download data from OpenJDK Vulnerability Group, and parse it into a more machine readable format .
+We use `ojvg_convert.py` to generate CycloneDX format objects, enhance with data from NIST, and generate a vdr, which is saved in `data/vdr.json`
+
+### Installation
+To install the requirements:
+
+`python3 -m pip install -r requirements.txt`
+
+
```
Comment (karianna) [inline README.md:14]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,20 @@
...
+### Tests
+There are some tests, in order to run them, you can do: 
+
+`python3 -m pytest`
+
+
```
Comment (karianna) [inline README.md:20]:
```suggestion
```

File: cvereporter/nist_enhance.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+from cyclonedx.model.impact_analysis import ImpactAnalysisAffectedStatus
+from cyclonedx.model.vulnerability import Vulnerability, VulnerabilitySource,VulnerabilityScoreSource, VulnerabilityRating, VulnerabilitySeverity, BomTarget, BomTargetVersionRange
+import requests
+import json
+
+"""
+this file has the utilities for downloading data about cves from NIST and updating Vulnerability objects with the data
+"""
+def fetch_nist(url: str, id: str) -> dict:
```
Comment (karianna) [inline cvereporter/nist_enhance.py:9]: Would be good to separate the def functions with some whitespace

**Review by karianna — CHANGES_REQUESTED** [review]

File: cvereporter/nist_enhance.py (hunk 3 above)
Comment (Scanteianu) [reply cvereporter/nist_enhance.py:9]: i'll run  a formatter

Comment (Scanteianu) [reply cvereporter/nist_enhance.py:19]: ran the formatter, added instructions for how to run

**Review by karianna — APPROVED** [review]

**Review by smlambert — APPROVED** [review]
Thanks @Scanteianu !!

Will merge this, then create several issues for next steps (setting up regular run, potentially integrating into overall release pipeline, etc).

**Pull request merged**